driver/ppm_events.c

```diff
@@ -636,12 +638,22 @@ int val_to_ring(struct event_filler_arguments *args, uint64_t val, u32 val_len,
 			/* Make sure the string is null-terminated */
 			if (likely(len > 0))
 			{
-				if (++len > (int)max_arg_size)
+				if(*(char *)(args->buffer + args->arg_data_offset + (len-1)) != '\0')
 				{
-					len = max_arg_size;
+					/* If we have not reached the `max_arg_size` we can put a new char with
+					 * the string terminator otherwise, we push the terminator on the last
+					 * available char without adding a new one (so no `len` increment).
```

Comment:
Wouldn't this mean we are overwriting a potentially valid character with `'\0'`? Is this acceptable? Leaving a string without the null terminator is way worse, but still...

Reply:
Yeah in some corner cases where we hit the maximum event dimension we have to overwrite the last char with a `\0` terminator...I think this is a good compromise because if we hit the max event len it means that we already truncate the param so truncating one more character to add the terminator shouldn't be a big issue, If we really reach this size `PPM_MAX_ARG_SIZE` i think it could be acceptable to truncate it  :thinking:  WDYT? @Molter73 @FedeDP 
```c
#define PPM_MAX_ARG_SIZE 65000
```

Reply:
Agree; i mean, it's better a small truncation rather than sending non-terminated string to userspace :D

Reply:
Yup, makes sense if we are already truncating.